src/movies.js

```diff
@@ -1,22 +1,77 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+// console.log('movies:', movies);
 
+// const movies = require('./data');
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+// const arrayDirector = [];
+function getAllDirectors(arrOfMovies) {
+  return (arrayDirector = arrOfMovies.map(function (element, value) {
+    // console.log(element);
+    return element.director;
+    // console.log(element);
+  }));
+  // return arrayDirector;
+}
+// getAllDirectors(movies);
+// console.log(arrayDirector);
+
+// const directorArray = getAllDirectors(movies);
+// console.log(directorArray);
+// console.log(arrayDirector);
+
+// console.log(arrayDirector);
+// console.log(arrayDirector);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(array) {
+  if (array.length === 0) {
+    return 0;
+  }
+  return (spielbergMovies = array.filter(function (element, value) {
+    if (
+      element.genre.includes('Drama') &&
+      element.director.includes('Steven Spielberg')
+    ) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
with filter method you do not need else (or return false)

If your condition passess -> filter will save those elements, if not passing it will ignore